src/models/match.js

```diff
@@ -101,6 +101,25 @@ export default () => {
     return await this.insertMany(mdocs);
   };
 
+  /**
+   * Given a list of allowed submissions _id, return all useable matches.
+   * @return { [MatchModel] }
+   * @param allowedSubmissionIds { [ObjectId] }
+   */
+  MatchSchema.statics.findByEffectiveSubmissionIds = async function(allowedSubmissionIds) {
+    const condition = allowedSubmissionIds.map(objectId => objectId.toString());
+    // DEBUG
+    console.log('condition:', condition);
+    return await this.find({
+      u1Submission: {
+        $in: condition,
+      },
+      u2Submission: {
+        $in: condition,
+      },
+    }).exec();
```

Comment:
I guess it is better to sort by `_id` desc because later matches will not overwrite results of front matches according to your code and we want a determinate result.